### ISSUE TITLE
Remove unnecessary conditions from code

### ISSUE THREAD

**Issue by jonkoops** [keycloak/keycloak-admin-ui#1128]
In various places there is code that has [unnecessary conditions](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md). This leads to a poor implementation and incorrect types and should not occur. 

The following needs to happen before this issue can be closed:
1. Refactor code that is flagged by the `@typescript-eslint/no-unnecessary-condition` rule so it no longer produces warnings (this can be split up in multiple PRs).
2. Change the same rule to  the `error` level so linting fails when unnecessary conditions are introduced.

**Comment by alevid99** [comment]
Hi @jonkoops to resolve one of the warnings related to the groupPickerDialog.
I need to know if is possible to move a child group to "root" to know how to refactor the code.
If this action should be possible a bugFix is needed otherwise we can simply disable the button.

Actually the button in the dialog is enabled but when you click the button nothing happen (you get a success message but you should get an error). On the confirmation callback the code crash (src/groups/GroupTable.tsx:214) the error get catch but due to the check src/groups/GroupTable.tsx:223 no error is raised.

**Comment by jonkoops** [comment]
Hmmm, yeah that sounds like it could be a separate issue. Could you create one and describe the problem there?

**Comment by alevid99** [comment]
I found this issue #1224 that track this problem. I can comment with more details there

**Comment by ssilvert** [comment]
> Hi @jonkoops to resolve one of the warnings related to the groupPickerDialog. I need to know if is possible to move a child group to "root" to know how to refactor the code. If this action should be possible a bugFix is needed otherwise we can simply disable the button.

Move to root should be possible.  So yes, it's a bug.
